waltz-ng/client/data-types/index.js

```diff
@@ -1,5 +1,28 @@
+import _ from "lodash";
+
 export default (module) => {
 
-    module.service('DataTypesDataService', require('./services/data-types-data'));
+    module.run(['$rootScope', '$state', 'DataTypesService', function($rootScope, $state, dataTypesService) {
+        $rootScope.$on('$stateChangeStart', function(evt, to, params) {
```

Comment:
I was expecting some feedback on this one.  My original thoughts were to
put this in common and allow the supply of a resolver function so it could
be generic and used everywhere.  Do you agree with that approach?

On Thu, Aug 11, 2016 at 5:31 PM, David Watkins <notifications@github.com>
wrote:

> In waltz-ng/client/data-types/index.js
> <https://github.com/khartec/waltz/pull/417#discussion_r74456282>:
>
> >  export default (module) => {
> >
> > -    module.service('DataTypesDataService', require('./services/data-types-data'));
> > +    module.run(['$rootScope', '$state', 'DataTypesService', function($rootScope, $state, dataTypesService) {
> > +        $rootScope.$on('$stateChangeStart', function(evt, to, params) {
>
> this *sounds* worryingly broad scoped..... Does it not apply to all state
> changes ?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/khartec/waltz/pull/417/files/e4bc887bdadc42f03d5e44e0876776ce30ea1a5c#r74456282>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AC4521yAyiyrMeOnHNNkjk73Z5dxbW_Xks5qe05JgaJpZM4JiBOI>
> .
>



-- 


*Kamran Saleem*
t  +44 (0) 7732 078654
e  kamran@thinkincode.co.uk
w thinkincode.co.uk